JamGame/Assets/Scripts/TileBuilder/Command.cs

```diff
@@ -9,7 +9,7 @@ namespace TileBuilder.Command
 {
     public interface ICommand
     {
-        public Result Execute(TileBuilderImpl tile_builder);
+        public Result Execute(TileBuilderImpl tile_builder, ref TileUnionImpl selected_tile);
```

Comment:
selected_tile don't used in all commands, so ICommand shouldn't know about it

Reply:
your new approach just replaces ref with wrapper class

Reply:
Don't be drunk, its not final commit

And selected_tile used in all commands except one

Reply:
https://github.com/mertwole/miniature-happiness/blob/931941b4ddd7a53cad063c4cb2154625b3966c1e/JamGame/Assets/Scripts/TileBuilder/Controller/Controller.cs#L16C9-L16C9

Reply:
Well, what don't you like?
It`s cover used not in all commands

Reply:
It's more about assumption that your API makes (that almost all operations are performed on selected tile). In my opinion it shouldn't be like this

Reply:
Actually, you don't really need selected tile at all, which will allow you to implement stateless API